src/main/java/seedu/address/model/tag/UniqueTagList.java

```diff
@@ -92,6 +130,34 @@ public void add(Tag toAdd) throws DuplicateTagException {
         assert CollectionUtil.elementsAreUnique(internalList);
     }
 
+    /**
+     * Remove a Tag from the list. The tag is identified by its name.
+     *
+     * throws TagNotFoundException if the Tag to remove is not found in the list.
+     * throws IllegalValueException if the Tag name input is invalid.
+     */
+    public Tag removeTag(String tagGettingRemoved) throws TagNotFoundException, IllegalValueException {
+        requireNonNull(tagGettingRemoved);
+        int tagIndexInList = contains(tagGettingRemoved);
+        if (tagIndexInList == -1) {
+            throw new TagNotFoundException("Tag is not found.");
+        } else {
+            return internalList.remove(tagIndexInList);
+        }
+    }
+
+    /**
+     * Remove a tag from the list.
+     * @param tagGettingRemoved
+     */
+    public void removeTag(Tag tagGettingRemoved) {
```

Comment:
Should use native List .remove(Object) implementation instead of re-implementing.